src/index.ts

```diff
@@ -1,132 +1,231 @@
-const assert = require("assert");
-
 export const OK_MARKER = 1;
 export const ERR_MARKER = 0;
 
-let codec: WebAssembly.Instance;
+let codec: WebAssembly.Instance
 
 // Init the library with the binary content of a .wasm file.
 // This function must be called once per client prior to use of any other package functions.
-export async function init(code: BufferSource) {
+export async function init(code: BufferSource) : Promise<void> {
     const wasm = await WebAssembly.compile(code);
     codec = await WebAssembly.instantiate(wasm, {})
 }
 
-export function isInitialized() {
-    return (codec !== null)
+// Returns true iff library was initialized with Wasm code
+export function isInitialized() : boolean {
+    return (codec !== undefined)
 }
 
-// Call an svm_codec function with the provided buffer. Returns result buffer.
-export function call(funcName: string, buf) {
-    return (codec!.exports as any)[funcName](buf)
+// User input to the encodeSpawnData function
+export interface Spawn {
+    version: number,
+    template: string,
+    name: string
+    ctor_name: string,
+    calldata: Uint8Array,
 }
 
-// Allocates a svm_codec buffer with the provided byte length
-export function wasmBufferAlloc(length: number) {
-    return (codec!.exports as any).wasm_alloc(length)
+// User input to the encodeCallData function
+export interface Call {
+    version: number,
+    target: string,
+    func_name: string,
+    verifydata: Uint8Array,
+    calldata: Uint8Array,
+}
+
+
+// Encodes the provided spawn app data
+export function encodeSpawn(data: Spawn) : Uint8Array {
```

Comment:
Both the `encodeSpawn`, `encodeCall` and `encodeInput` behave the same
(`wasm_encode_spawn` vs `wasm_encode_call` vs `wasm_encode_inputdata`)

I think you can extract both into a common abstraction. 
Maybe calling it `encodeMessage` would make a good naming.

Later in the review, I've given a similar comment for the decoding side.